portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/handlers/SiteHandler.java

```diff
@@ -1224,5 +1245,11 @@ public boolean doToolBuffer(HttpServletRequest req, HttpServletResponse res,
 
 		return true;
 	}
+	
+	public void resetAttributte(Session session) {
+		session.setAttribute(ALLOW_DIRECT_LINK, null);
+		session.setAttribute(DIRECT_LINK_SITE, null);
+		session.setAttribute(DIRECT_LINK_PAGE, null);
+	}
```

Comment:
Deleted.
These attributes were to avoid that the student access the hidden tool apart from the direct link on the browser.
They depend on how we understand this feature.

Thank you, Adrian.